### PR TITLE
Solution react_lists-of-goods-hooks

### PR DESCRIPTION
**Pull request by NazariiAlieksieiev**
[DEMO LINK](https://NazariiAlieksieiev.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +15,101 @@ export const goodsFromServer = [
...
+        {sortType !== SortType.NONE || isReversed ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              return (
+                setSortType(SortType.NONE),
+                setReversMode(false)
+              );
+            }}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:99]:
```suggestion
            onClick={() => {
                setSortType(SortType.NONE),
                setReversMode(false)
            }}
```
redundant return
it is bad practice to make anonymous arrow functions in JSX.
make function in compnent and pass reference here

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +15,101 @@ export const goodsFromServer = [
...
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // Sort and reverse goods if needed
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:53]: remove console

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +15,101 @@ export const goodsFromServer = [
...
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+  const [isReversed, setReversMode] = useState(false);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortType !== SortType.ALPHABET && 'is-light'}`}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:67]: consider to use [classnames library](https://www.npmjs.com/package/classnames) for building classes dynamically

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +15,101 @@ export const goodsFromServer = [
...
+  // Sort and reverse goods if needed
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+  const [isReversed, setReversMode] = useState(false);
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:60]:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
it is general convention of naming state and setter of state

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Good job.
Need some little improvements

File: src/App.tsx (hunk 3 above)
Comment (etojeDenys) [reply src/App.tsx:68]: still not fixed

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**